Book.java

```diff
@@ -0,0 +1,315 @@
+package accounting;
+
+import agents.Agent;
+import actions.Action;
+import contracts.Asset;
+import contracts.Contract;
+import contracts.Loan;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * This is the main class implementing double entry accounting. All public operations provided by this class
+ * are performed as a double entry operation, i.e. a pair of (dr, cr) operations.
+ *
+ * A Book contains a set of accounts, and is the interface between an agent and its accounts. Agents cannot
+ * directly interact with accounts other than via a Book.
+ *
+ * At the moment, a Book contains an account for each type of contract, plus an equity account and a cash account.
+ *
+ * A simple economic agent will usually have a single Book, whereas complex firms and banks can have several books
+ * (as in branch banking for example).
+ *
+ * @author rafa
+ */
+public class Book implements BookAPI {
```

Comment:
Is 'Book' intended to be a parent class to most Books, or an example of using the BookAPI?